### PR TITLE
CA-249084

### PR DESCRIPTION
**Pull request by GaborApatiNagy**

### CODE REVIEW

File: XenAdmin/Wizards/DRWizards/DRFailoverWizardStoragePage.cs [hunk 1]
```diff
@@ -332,8 +332,7 @@ private List<FibreChannelDevice> FiberChannelScan()
 
             Host master = Helpers.GetMaster(Connection);
 
-            if (master != null && (Helpers.HostBuildNumber(master) >= 9633
-                                || Helpers.HostBuildNumber(master) == Helpers.CUSTOM_BUILD_NUMBER))
+            if (master != null)
```
Comment (MihaelaStoica) [inline XenAdmin/Wizards/DRWizards/DRFailoverWizardStoragePage.cs]: We don't need the master check anymore and we should remove the `dconf[SrProbeAction.DEVICE] = device.Path;` completelly. That was only applicable for old versions.

File: XenAdmin/Wizards/NewSRWizard_Pages/SrWizardType.cs [hunk 2]
```diff
@@ -62,8 +62,7 @@ public LvmOhbaSrDescriptor(FibreChannelDevice device, IXenConnection connection)
 
             // CA-19025: Change XenCenter SR.create for LVMoHBA to use the
             // updated SCSIid parameter rather than the device path
-            if (master != null && (Helpers.HostBuildNumber(master) >= 9633
-                                   || Helpers.HostBuildNumber(master) == Helpers.CUSTOM_BUILD_NUMBER))
+            if (master != null)
```
Comment (MihaelaStoica) [inline XenAdmin/Wizards/NewSRWizard_Pages/SrWizardType.cs]: As above, we don't need the old code with device path.

**Comment by MihaelaStoica** [comment]
We should also fix the use of HostBuildNumber on the MainWindow.cs. Although it is just a log entry, we should use the build number as it is, without trying to convert it to integer first.
Actually, we should remove all usages of that function, and use string matching instead, where needed.

File: XenAdmin/Wizards/DRWizards/DRFailoverWizardStoragePage.cs [hunk 3]
```diff
@@ -332,15 +332,10 @@ private List<FibreChannelDevice> FiberChannelScan()
...
```
Comment (MihaelaStoica) [inline XenAdmin/Wizards/DRWizards/DRFailoverWizardStoragePage.cs]: I think we don't need to resolve the master here. If the master is null, we have greater problems than the SCSIId being added to the DeviceConfig.

**Comment by MihaelaStoica** [comment]
@GaborApatiNagy, could you please also resolve the merge conflicts with master?

File: XenAdmin/Wizards/DRWizards/DRFailoverWizardStoragePage.cs (hunk 3 above)
Comment (GaborApatiNagy) [inline XenAdmin/Wizards/DRWizards/DRFailoverWizardStoragePage.cs]: 🤣 true... I'll remove that

File: XenModel/XenAPI-Extensions/Host.cs [hunk 4]
```diff
@@ -713,11 +713,15 @@ public virtual string PlatformVersion
...
+        /// From Falcon, this property is not used.
+        /// </summary>
+        /// <remarks>
         /// Return the build number of this host, or -1 if none can be found.  This will often be
         /// 0 or -1 for developer builds, so comparisons should generally treat those numbers as if
         /// they were brand new.
-        /// </summary>
-        public int BuildNumber
+        /// </remarks>
+        internal int BuildNumber
```
Comment (GaborApatiNagy) [inline XenModel/XenAPI-Extensions/Host.cs]: good idea

File: XenAdmin/Wizards/NewSRWizard_Pages/SrWizardType.cs [hunk 5]
```diff
@@ -62,15 +62,10 @@ public LvmOhbaSrDescriptor(FibreChannelDevice device, IXenConnection connection)
...
```
Comment (MihaelaStoica) [inline XenAdmin/Wizards/NewSRWizard_Pages/SrWizardType.cs]: We should remove the master check from here as well

**Comment by MihaelaStoica** [comment]
Looks good now. Merging.

**Pull request merged**